### ISSUE TITLE
Stack underflows panics inputcircuitbuilder

### ISSUE THREAD

**Issue by adria0** [appliedzkp/zkevm-circuits#506]
Trying to prove an `ADD STOP` circuit panics `thread 'main' panicked at 'attempt to subtract with overflow', eth-types/src/evm_types/stack.rs:114:18`. It should revert.

**Comment by barryWhiteHat** [comment]
I don't think we have handled these kind of exceptional reverts yet. At the moment our circuit will not allow us ot make a proof if this happens. 

The way i look at this is that its on our road to 100% evm compatibility. We will get some basic opcodes working and then add the feature that stack under flows revert.

**Comment by adria0** [comment]
Cool, I'm trying to pass as much "ethereum offical tests" as we can, disabling the ones that are still not implemented.

**Comment by ed255** [comment]
Ah, this seems to be an underflow in Rust arithmetic, which is a bug in our code; we need to fix that.

And then I guess we need to make sure we are able to correctly calculate the circuit inputs with traces that contain error situations (like stack overflow / underflow); which it should!

**Comment by CPerezz** [comment]
There are functions to do `wrapped_sub` and similar stuff so that we can prevent underflows. But as said avobe. We can leave this kind of revert handling cases for the near future once the entire solution is built.
